share/availability/discovery/discovery.go

```diff
@@ -57,6 +61,19 @@ func NewDiscovery(
 		peersLimit,
 		discInterval,
 		advertiseInterval,
+		nil,
+	}
+}
+
+func (d *Discovery) WithOnPeersUpdate(f OnPeersUpdate) {
+	if d.onPeersUpdate == nil {
+		d.onPeersUpdate = f
+		return
+	}
+
+	d.onPeersUpdate = func(peers []peer.ID) {
+		d.onPeersUpdate(peers)
```

Comment:
why do we need to call `d.onPeersUpdate(peers)` inside assigning to `d.onPeersUpdate`?